File: src/main/java/com/naminhyeok/fantazzk/room/RoomException.java
package com.naminhyeok.fantazzk.room;

import com.naminhyeok.fantazzk.CoreException;
import com.naminhyeok.fantazzk.template.TemplateId;
import java.util.Map;

public class RoomException extends CoreException {
    private RoomException(RoomErrorType error, Object data) {
        super(error, data);
    }

    public static RoomException notFound(String code) {
        return new RoomException(RoomErrorType.ROOM_NOT_FOUND, Map.of("code", code));
    }

    public static RoomException templateNotFound(TemplateId templateId) {
        return new RoomException(
            RoomErrorType.ROOM_TEMPLATE_NOT_FOUND,
            Map.of("templateId", templateId.templateId().toString())
        );
    }

    public static RoomException notJoinable(String detail) {
        return new RoomException(RoomErrorType.ROOM_NOT_JOINABLE, Map.of("detail", detail));
    }

    public static RoomException roomFull() {
        return new RoomException(RoomErrorType.ROOM_FULL, Map.of("detail", RoomErrorType.ROOM_FULL.message()));
    }

    public static RoomException notStartable(String detail) {
        return new RoomException(RoomErrorType.ROOM_NOT_STARTABLE, Map.of("detail", detail));
    }
}

File: room/exception/src/main/kotlin/com/naminhyeok/fantazzk/room/exception/RoomException.kt
package com.naminhyeok.fantazzk.room.exception

sealed class RoomException(
    val errorCode: String,
    message: String,
) : RuntimeException(message) {
    class RoomNotFoundException : RoomException(
        errorCode = "ROOM_NOT_FOUND",
        message = "방을 찾을 수 없습니다",
    )

    class TeamLeaderNotFoundException : RoomException(
        errorCode = "ROOM_TEAM_LEADER_NOT_FOUND",
        message = "팀장을 찾을 수 없습니다",
    )
}
